agent/command/s3_put.go

```diff
@@ -550,9 +550,11 @@ func (s3pc *s3put) attachFiles(ctx context.Context, comm client.Communicator, lo
 		if s3pc.Visibility == artifact.Signed {
 			bucket = s3pc.Bucket
 			fileKey = remoteFileName
-			// TODO (DEVPROD-13658): Check if the bucket is internal and use the app's server IRSA credentials.
-			key = s3pc.AwsKey
-			secret = s3pc.AwsSecret
+			// If the bucket is not an internal one, use the provided credentials.
```

Comment:
Might help to clarify the opposite case where a user provides an internally-owned buckets. Something about how Evergreen doesn't need credentials from the task for pre-signing files because it already has permission to access internal buckets.